app/presenters/user_presenter.rb

```diff
@@ -0,0 +1,31 @@
+class UserPresenter < SimpleDelegator
+  include ActionView::Helpers::FormOptionsHelper
+
+  delegate :global, to: :user_roles, prefix: true
+
+  def self.wrap(users)
+    users.map { |user| new(user) }
```

Comment:
I would think you would return some sort of lazy object that would only instantiate the `map` when some other enumerable was called on it. Something like this. Not tested at all, and probably not worth bringing in right now.

```ruby
class DecoratedCollection
  attr_reader :collection, :decorator

  delegate :each, to: :collection
  include Enumerable

  def initialize(collection, decorator)
    @collection = collection
    @decorator = decorator
  end

  def wrapped_collection
    @wrapped_collection ||= collection.map { |obj| decorator.new(obj) }
  end
end
```
This is not unlike what Draper does: https://github.com/drapergem/draper/blob/master/lib/draper/collection_decorator.rb